clients/src/test/java/org/apache/kafka/test/MockConsumerInterceptor.java

```diff
@@ -73,19 +73,24 @@ public ConsumerRecords<String, String> onConsume(ConsumerRecords<String, String>
         CLUSTER_ID_BEFORE_ON_CONSUME.compareAndSet(NO_CLUSTER_ID, CLUSTER_META.get());
 
         Map<TopicPartition, List<ConsumerRecord<String, String>>> recordMap = new HashMap<>();
+        Map<TopicPartition, OffsetAndMetadata> nextOffsets = new HashMap<>();
+
         for (TopicPartition tp : records.partitions()) {
             List<ConsumerRecord<String, String>> lst = new ArrayList<>();
+            long nextOffset = 0;
```

Comment:
Have you considered the case here where the records for the topic partition is empty? Won't you return "next offset 0" in this case?